src/libpmemfile-posix/lseek.c

```diff
@@ -231,6 +231,12 @@ add_off(size_t cur, pmemfile_off_t off)
 {
 	if (off >= 0)
 		return (pmemfile_off_t)(cur + (size_t)off);
+
+	/* -INT64_MIN can't be represented in pmemfile_off_t */
+	COMPILE_ERROR_ON(sizeof(pmemfile_off_t) != sizeof(int64_t));
+	if (off == INT64_MIN)
```

Comment:
I still say it would simpler/ cleaner:
```
size_t x = cur + (size_t)off;
if (x < cur || x > PMEMFILE_OFF_T_MAX)
   return -1;
else
   return (pmemfile_off_t)x;
```
No implementation defined, or undefined behaviour.

Reply:
Ah, right, except for the first cast.

Reply:
x can be smaller than cur when off is negative, so it's not that simple.

Reply:
Ok, with just x < cur removed I think it's correct. I'll post new version in a minute.